org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenExecutionContext.java

```diff
@@ -331,6 +332,7 @@ private static void executeMojo(MavenSession session, MojoExecution execution, I
       lookup.lookup(BuildPluginManager.class).executeMojo(session, execution);
     } catch(Exception ex) {
       session.getResult().addException(ex);
+      throw new CoreException(Status.error("Failed to execute mojo", ex));
```

Comment:
Thats not the place to handle this, the caller has to check the result and act accordingly (e.g. m2e has helper classes to create error markers from results).

Reply:
I checked this more deeply in m2e an it seems this actually is already handled at a higher level, even though in some places we return the result. It seems strange here to have both (throwing an exception and adding it to the result). So I think simply throwing an exception would be enough